File: leetcode-solutions/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}

        for i, value in enumerate(nums):

            complement = target - value

            if complement in seen:
                return [i, seen[complement]]
            
            else:
                seen[value] = i

        # O(n^2) Solution:

        # for i in range(len(nums)):
        #     for j in range(i+1, len(nums)):  # start at i+1 to avoid duplicate pairs
        #         if nums[i] + nums[j] == target:
        #             return [i, j]

File: leetcode-solutions/0001-two-sum/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut seen = HashMap::new();

        for (i, &value) in nums.iter().enumerate() {
            let remaining = target - value;
            if let Some(&j) = seen.get(&remaining) {
                return vec![i as i32, j as i32];
            }
            seen.insert(value, i);
        }

        vec![] // return an empty vector if no solution is found
    }
}

